File: WebArticles.WebAPI/Data/Models/UserModel.cs
﻿using DataModel.Data.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WebArticles.WebAPI.Data.Models
{
    public class UserModel
    {
        public long Id { get; set; }

        public string ProfilePickLink { get; set; }

        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Username { get; set; }
        public string Email { get; set; }
        public DateTime? BirthDate { get; set; }

        public int WriterRating { get; set; }
        public string WriterDescription { get; set; }
        public Topic[] WriterTopics { get; set; }

        public int ReviewerRating { get; set; }
        public string ReviewerDescription { get; set; }
        public Topic[] ReviewerTopics { get; set; }

    }
}

File: WebArticles.Angular/src/app/data-model/models/user.model.ts
import { Topic } from "./topic.model";

export class User {
    id: number;
    profilePickLink: string;
    firstName: string;
    lastName: string;
    userName: string;
    email: string;
    birthDate: Date;

    writerRating: number;
    writerDescription: string;
    writerTopics: Topic[];

    reviewerRating: number;
    reviewerDescription: string;
    reviewerTopics: Topic[];
}
